Service/InstallationService.php

```diff
@@ -827,6 +836,38 @@ private function createEndpoints(array $endpointsData = []): array
         return $endpoints;
     }//end createEndpoints()
 
+    /**
+     * This function handles the creation of an subEndpoint by adding extra data from the $subEndpointsConfig to the already created $subEndpoint with basic data.
+     *
+     * @param Endpoint $subEndpoint        A newly created endpoint with some basic data.
+     * @param array    $subEndpointsConfig Configuration from the installation.json ['endpoints']['subEndpoints'] array.
+     *
+     * @return Endpoint The updated subEndpoint with data from the $subEndpointsConfig.
+     */
+    private function handleSubEndpointsConfig(Endpoint $subEndpoint, array $subEndpointsConfig): Endpoint
+    {
+        if (isset($subEndpointsConfig['path'])) {
+            $this->logger->debug('Creating a subEndpoint '.$subEndpointsConfig['path'].' for '.$subEndpoint->getEntity()->getReference());
+
+            $path = $subEndpoint->getPath();
+            $path[] = $subEndpointsConfig['path'];
+            $subEndpoint->setPath($path);
+            $subEndpoint->setPathRegex($subEndpoint->getPathRegex().'/'.$subEndpointsConfig['path']);
+            $subEndpoint->setName($subEndpoint->getName().' '.ucfirst($subEndpointsConfig['path']));
+
+            if (isset($subEndpointsConfig['description']) === true) {
+                $subEndpoint->setDescription($subEndpointsConfig['description']);
+            }
+            if (isset($subEndpointsConfig['throws']) === true) {
+                $subEndpoint->setThrows($subEndpointsConfig['throws']);
+            }
+        } else {
+            $this->logger->error('SubEndpointsConfig is missing a path', ['SubEndpointsConfig' => $subEndpointsConfig]);
+        }
+
+        return $subEndpoint;
+    }//end handleSubEndpointsConfig()
+
```

Comment:
Uhm, this sounds fine, but shouldn't this be in a separate PR?

Reply:
Yess it could and maybe should be. but we will need it for DIT-KISS just like the other code. It won't break anything, only used / touched by the ZGW-bundle as soon as that is merged/tested